pkg/injector/bootstrap_test.go

```diff
@@ -318,3 +331,159 @@ var _ = Describe("Test functions creating Envoy bootstrap configuration", func()
+
+	testCases := []struct {
+		name       string
+		secrets    []*corev1.Secret
+		expSecrets []*corev1.Secret
+	}{
+		{
+			name: "get bootstrap secrets from k8s secrets",
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      constants.DefaultCABundleSecretName,
+						Namespace: "testNamespace",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      bootstrapSecretPrefix + "proxyUUID",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+			expSecrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      bootstrapSecretPrefix + "proxyUUID",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+		},
+		{
+			name: "no bootstrap secrets in k8s secrets",
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      constants.DefaultCABundleSecretName,
+						Namespace: "testNamespace",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notBootstrapSecret",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+			expSecrets: []*corev1.Secret{},
+		},
+	}
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Running test case %d: %s", i, tc.name), func(t *testing.T) {
+			assert := tassert.New(t)
+			informerCollection, err := informers.NewInformerCollection("testNamespace", nil, informers.WithKubeClient(fakeClient))
+			assert.Nil(err)
+
+			b := NewBootstrapSecretRotator(context.Background(), fakeClient, informerCollection, fakeCertManager, time.Duration(1))
+			for _, s := range tc.secrets {
+				err = informerCollection.Add(informers.InformerKeySecret, s, t)
+				assert.Nil(err)
+			}
+
+			actual := b.listBootstrapSecrets()
+			assert.ElementsMatch(tc.expSecrets, actual)
+		})
+	}
+}
+
+func TestRotateBootstrapSecrets(t *testing.T) {
+	assert := tassert.New(t)
+
+	testNs := "testNamespace"
+	proxyUUID := uuid.New().String()
+
+	testCases := []struct {
+		name     string
+		certName string
+		secrets  []*corev1.Secret
+	}{
+		{
+			name:     "update bootstrap secret with new cert",
+			certName: certificate.CommonName(proxyUUID + ".test.cert").String(),
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      bootstrapSecretPrefix + proxyUUID,
+						Namespace: testNs,
+					},
+					Data: map[string][]byte{
+						"ca.crt":  pem.RootCertificate("zz"),
+						"tls.crt": pem.Certificate("xx"),
+						"tls.key": pem.PrivateKey("yy"),
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notBootstrapSecret",
+						Namespace: testNs,
+					},
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Running test case %d: %s", i, tc.name), func(t *testing.T) {
+			getCertValidityPeriod := func() time.Duration { return 5 * time.Second }
+			certManager, err := certificate.NewManager(context.Background(), &certificate.FakeMRCClient{}, getCertValidityPeriod, getCertValidityPeriod, 5*time.Second)
```

Comment:
you can also just call certificate.FakeCertManager() and keep fakeMRCClient unexposed